### PR TITLE
Add disable-infra annotation to hive clusterdeployments

### PR DESCRIPTION
**Pull request by tsatam**
### Which issue this PR addresses:

Fixes [ARO-11607](https://issues.redhat.com/browse/ARO-11607)

### What this PR does / why we need it:

Adds the `hive.openshift.io/infra-disabled` annotation to our Hive ClusterDeployments (for both new installs as well as adopted clusters), set to `true`. This is required in order to disable the hibernation controller, which does not currently work for ARO MIWI clusters. 

> [!NOTE]
> This PR currently disables the hibernation controller for all cluster installations (CSP and MIWI) - we do not use this functionality in ARO's implementation of Hive today. 

See corresponding change in Hive to add this annotation for more details: https://github.com/openshift/hive/pull/2501 

### Test plan for issue:

- [x] E2E test was added to ensure the annotation is set
- [x] E2E cluster creation ensures the cluster is installed successfully with this annotation set

### Is there any documentation that needs to be updated for this PR?

No

### How do you know this will function as expected in production? 

E2E tests outlined above

### CODE REVIEW

File: pkg/hive/resources.go [hunk 1]
```diff
@@ -193,6 +193,11 @@ func adoptedClusterDeployment(namespace, clusterName, clusterID, infraID, locati
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      ClusterDeploymentName,
 			Namespace: namespace,
+			Annotations: map[string]string{
+				// https://github.com/openshift/hive/pull/2501
+				// Disable hibernation controller
```
Comment (cadenmarchese) [inline pkg/hive/resources.go:198]: Tiniest nit, but it's not clear looking at the code or the linked PR why we're disabling it.
```suggestion
				// Disable hibernation controller as it is not used as part of ARO's Hive implementation 
```

**Review by cadenmarchese — APPROVED** [review]

File: pkg/hive/resources.go [hunk 2]
```diff
@@ -193,6 +193,11 @@ func adoptedClusterDeployment(namespace, clusterName, clusterID, infraID, locati
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      ClusterDeploymentName,
 			Namespace: namespace,
+			Annotations: map[string]string{
+				// https://github.com/openshift/hive/pull/2501
+				// Disable hibernation controller as it is not used as part of ARO's Hive implementation 
+				"hive.openshift.io/infra-disabled": "true",
```
Comment (kimorris27) [inline pkg/hive/resources.go:199]: Super nit: we could make a `const` for this in `pkg/hive` since it's used in two places.

**Review by kimorris27 — COMMENTED** [review]
LGTM. I only left my nit because there's a formatting issue that needs to be fixed anyway.

File: pkg/hive/resources.go (hunk 2 above)
Comment (tsatam) [reply pkg/hive/resources.go]: addressed in latest commit

File: pkg/hive/install.go [hunk 3]
```diff
@@ -214,6 +214,10 @@ func (c *clusterManager) clusterDeploymentForInstall(doc *api.OpenShiftClusterDo
 
 				// TODO: remove until we use a version of hive at minimal install
 				"hive.openshift.io/cli-domain-from-installer-image": "true",
+
+				// https://github.com/openshift/hive/pull/2501
+				// Disable hibernation controller
+				"hive.openshift.io/infra-disabled": "true",
```
Comment (kimorris27) [inline pkg/hive/install.go:220]: One more super nit since you humored me on the first one: you could use `hiveInfraDisabledAnnotation` here too. 🙇🏻

Comment (tsatam) [reply pkg/hive/install.go:220]: that's what I get for making this change live during PR review session and not fully paying attention. I'll fix this.

**Review by cadenmarchese — APPROVED** [review]

**Review by kimorris27 — APPROVED** [review]

**Pull request merged**